src/main/java/net/dv8tion/jda/internal/entities/GuildImpl.java

```diff
@@ -1153,6 +1153,23 @@ public List<GuildVoiceState> getVoiceStates()
                 .collect(Helpers.toUnmodifiableList());
     }
 
+    @Nonnull
+    @Override
+    @CheckReturnValue
+    public RestAction<GuildVoiceState> retrieveMemberVoiceStateById(long id)
+    {
+        JDAImpl jda = getJDA();
+        Route.CompiledRoute route = Route.Guilds.GET_VOICE_STATE.compile(getId(), Long.toUnsignedString(id));
+        return new RestActionImpl<>(jda, route, (response, request) ->
+        {
+            EntityBuilder entityBuilder = jda.getEntityBuilder();
+            DataObject voiceStateData = response.getObject();
+            MemberImpl member = entityBuilder.createMember(this, voiceStateData.getObject("member"), voiceStateData, null);
```

Comment:
I would not pass the voice state to `createMember` here since it will become stale and inconsistent.